pkg/virt-handler/node-labeller/node_labeller.go

```diff
@@ -0,0 +1,283 @@
+func (n *NodeLabeller) Run(threadiness int, stop chan struct{}) {
+	defer n.queue.ShutDown()
+	n.logger.Infof("node-labeller is running")
+
+	err := n.loadCPUInfo()
+	if err != nil {
+		n.logger.Errorf("node-labeller could not load cpu info: " + err.Error())
+	}
+
+	err = n.loadHostCapabilities()
+	if err != nil {
+		n.logger.Errorf("node-labeller could not load host capabilities: " + err.Error())
+	}
+
+	n.loadHypervFeatures()
+
+	n.clusterConfig.SetConfigModifiedCallback(func() {
+		err := n.run()
+		if err != nil {
+			n.logger.Errorf("node-labeller cannot update labels %s", err.Error())
+		}
+	})
+
+	n.nodeInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: func(_, _ interface{}) {
+			n.queue.Add(n.host)
+		},
+	})
+
+	go n.nodeInformer.Run(stop)
+	cache.WaitForCacheSync(stop, n.nodeInformer.HasSynced)
+
+	for i := 0; i < threadiness; i++ {
+		go wait.Until(n.runWorker, time.Second, stop)
+	}
+	<-stop
+}
+func (n *NodeLabeller) runWorker() {
+	for n.execute() {
+	}
+}
+
+func (n *NodeLabeller) execute() bool {
+	key, quit := n.queue.Get()
+	if quit {
+		return false
+	}
+	defer n.queue.Done(key)
+
+	if !n.kvmController.NodeHasDevice(kvmPath) {
+		n.logger.Errorf("node-labeller cannot work without KVM device.")
+		return true
+	}
+
+	err := n.run()
+
+	if err != nil {
+		n.logger.Errorf("node-labeller sync error encountered: %v", err)
+		n.queue.AddRateLimited(key)
+	} else {
+		n.queue.Forget(key)
+	}
+	return true
+}
+
+func (n *NodeLabeller) run() error {
+	var (
+		cpuFeatures map[string]bool
+		cpuModels   []string
+	)
+
+	//parse all informations
+	cpuModels, cpuFeatures = n.getCPUInfo()
+
+	//load node
+	nodeObj, exists, err := n.nodeInformer.GetStore().GetByKey(n.host)
+	if err != nil || !exists {
+		n.logger.Infof("node-labeller cannot get node %s", n.host)
+		return err
+	}
+
+	var (
+		ok           bool
+		originalNode *v1.Node
+	)
+
+	if originalNode, ok = nodeObj.(*v1.Node); !ok {
+		n.logger.Infof("node-labeller cannot convert node " + n.host)
+		return fmt.Errorf("Could not convert node " + n.host)
+	}
+
+	node := originalNode.DeepCopy()
+
+	//prepare new labels
+	newLabels := n.prepareLabels(cpuModels, cpuFeatures)
+	//remove old labeller labels
+	n.removeLabellerLabels(node)
+	//add new labels
+	n.addLabellerLabels(node, newLabels)
+
+	//patch node only if there is change in labels or annotations
+	if !reflect.DeepEqual(originalNode.Labels, node.Labels) || !reflect.DeepEqual(originalNode.Annotations, node.Annotations) {
+		err = n.patchNode(originalNode, node)
+	}
+	return err
+}
+
+func (n *NodeLabeller) patchNode(originalNode, node *v1.Node) error {
+	type payload struct {
+		Op    string            `json:"op"`
+		Path  string            `json:"path"`
+		Value map[string]string `json:"value"`
+	}
+
+	p := []payload{
+		{
+			Op:    "test",
+			Path:  "/metadata/labels",
+			Value: originalNode.Labels,
+		}, {
+			Op:    "replace",
+			Path:  "/metadata/labels",
+			Value: node.Labels,
+		},
+	}
```

Comment:
you're performing a deepEquals check on the annotations below. do we need the same thing for the labels?